test/embind/test_val.cpp

```diff
@@ -384,15 +391,15 @@ int main() {
 
   test("template<typename... Args> val new_(Args&&... args)");
   EM_ASM(
-    A = function() {
+    globalThis.A = function() {
```

Comment:
Is this needed because A is more local in ESM mode?

Reply:
Its just because in strict mode A needs to be declared somewhere.  In non-strict mode you can just do `A = 10` and it will create `A` for you.

Do i need to use the global scope here.. let me check.

Reply:
Yes the old code was relying on the fact that doing `A = 10` implicitly creates a global called `A`.  That doesn't work in strict mode.

i.e. in non-strict mode you can do this:

```
function doit () {
  // Creates a global A
  A = 10;
}

doit();
console.log(A);
```